indico/modules/events/client/js/header.jsx

```diff
@@ -8,18 +8,13 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 
-import {IButton, ICSCalendarLink} from 'indico/react/components';
+import {ICSCalendarLink} from 'indico/react/components';
 import {Translate} from 'indico/react/i18n';
 
-document.addEventListener('DOMContentLoaded', () => {
-  const calendarContainer = document.querySelector('#event-calendar-link');
+let calendarExportContainer = null;
 
-  if (!calendarContainer) {
-    return;
-  }
-
-  const {eventId, eventContribCount, eventSessionBlockCount} = calendarContainer.dataset;
-  const eventInSeries = calendarContainer.dataset.eventInSeries !== undefined;
+document.addEventListener('indico:action:calendar-export', e => {
```

Comment:
The very first thing I noticed when testing this is that it suffers from the usual issue: Clicking the button again while the popup is open, closes and immediately reopens it, instead of just closing it.